### DISCUSSION TITLE
提议: 每一个组件单独发布一个包?

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#43864] [Ideas]
用来解决的问题: 
// before
rc-select 组件,通过 pnpm patch 改动源码, 导致整个 antd 无法升级

//after

```jsx
import Select from 'antd/select' // 单独发包

import {Button} from 'antd' // 照旧
```

这样就能化解这个问题, 而且你们用了 cssinjs ,搭配分包策略是水到渠成的事情 , 你们觉得可行吗?

**Comment by zombieJ** [comment]
先说结论，不会这么做。

* 组件本身是有相互依赖的，例如 TreeSelect、Cascader 依赖 Select，Table 依赖 Pagination、Select、Button、Dropdown 等等。
* 不同的 npm 对 node_modules 管理方式不同，甚至不同版本对依赖提升的逻辑也不同，针对某个 npm 做 patch 不是推荐的做法。
* 独立包本身依赖的还是相同的 rc 组件，魔改一个也可能会影响其他依赖于 rc 的组件库。

最后，避免 XY 问题。很多情况下，HOC 都能解决。如果没有透出来，最好的方式是 fork rc-select 然后自己加个样式发个包。

**Reply by vaynevayne** [reply]
关于第一点: 有依赖也不影响单独发包
关于第二点: 对于业务项目来说,只使用一个 npm 工具 也是能接受的, 这是成本最小的方式
关于第三点, pnpm 貌似支持指定包路径

```jsx
  "patchedDependencies": {
      "rc-select@14.5.1": "patches/rc-select@14.5.1.patch",
        "antd/select/rc-select@14.5.1": "patches/rc-select@14.5.1.patch", // 应该支持这样的
    }
```

**Comment by vaynevayne** [comment]
rc-select 然后自己加个样式发个包。 这个有什么 demo 吗? 有没有一行代码支持使用 antd 的样式?

**Reply by RedJue** [reply]
可以直接使用v5的token体系来接入，demo 直接看 antd 源码就行，大部分组件都是基于 rc 封装的。

**Reply by vaynevayne** [reply]
代码太多了,看不到核心啊, 另外还会有些逻辑在 antd 里吧,岂不是除了 token,还要这些逻辑也要搬

**Reply by RedJue** [reply]
antd 里的逻辑是为了 antd 组件需要才加的，rc 的功能已经是完整的，如果不需要额外功能，直接改样式就行
